cdap-ui/app/features/hydratorplusplus/controllers/create/toppanel-ctrl.js

```diff
@@ -157,9 +157,124 @@ class HydratorPlusPlusTopPanelCtrl{
   showSettings() {
     this.state.viewSettings = !this.state.viewSettings;
   }
-}
 
-HydratorPlusPlusTopPanelCtrl.$inject = ['$stateParams', 'HydratorPlusPlusConfigStore', 'HydratorPlusPlusConfigActions', '$uibModal', 'HydratorPlusPlusConsoleActions', 'DAGPlusPlusNodesActionsFactory', 'GLOBALS', 'myHelpers', 'HydratorPlusPlusConsoleStore', 'myPipelineExportModalService', '$timeout', '$scope'];
+  // PREVIEW
+  startTimer() {
+    this.previewTimerInterval = this.$interval(() => {
+      let duration = (new Date() - this.previewStartTime) / 1000;
```

Comment:
We could avoid the checks and manual construction of time. If you don't feel its going to add much then thats ok. I don't have really a strong opinion.